src/PrestaShopBundle/Routing/LegacyRouterChecker.php

```diff
@@ -116,6 +120,13 @@ public function check(Request $request): bool
 
             $controllerClass = $controllers[strtolower($queryController)];
         }
+        // Loading controller
```

Comment:
```suggestion
        // We load the controller early in the process (during router matching actually), because the controller
        // configuration has many impacts on the contexts, the security listeners, ... And the relevant data can
        // only be retrieved once the legacy class is instantiated to access its public configuration
        // But for performance issues we only instantiate (and init) the controller here once and then store it (along
        // with other related attributes) in the request attributes so they can be retrieved easily by the code depending on them
```